dalm/pipelines/reading_comprehension_pipeline.py

```diff
@@ -90,28 +94,28 @@ def pipeline(
     # mix both and make it a huggingface dataset
     list_of_data = []
     if comprehension_type in [SynthMode.REGEX, SynthMode.BOTH]:
-        a1 = load_dataset("json", data_files=regex_dataset_output_path)
-        # for file in os.listdir(regex_dataset_output_path):
-        #    text = open(file, 'r').read()
-        #    list_of_data.append(text)
+        # a1 = load_dataset("json", data_dir=regex_dataset_output_path) This does not work
+        for file in os.listdir(regex_dataset_output_path):
+            text = json.load(open(os.path.join(regex_dataset_output_path, file), "r"))
+            list_of_data.append({"messages": text})
 
     if comprehension_type in [SynthMode.LLM, SynthMode.BOTH]:
-        a2 = load_dataset("json", data_files=llm_dataset_output_path)
-        # for file in os.listdir(llm_dataset_output_path):
-        #    text = open(file, 'r').read()
-        #    list_of_data.append(text)
+        # a2 = load_dataset("json", data_dir=llm_dataset_output_path) This does not work
+        for file in os.listdir(llm_dataset_output_path):
+            text = json.load(open(os.path.join(llm_dataset_output_path, file), "r"))
+            list_of_data.append({"messages": text})
 
     if comprehension_type == SynthMode.BOTH:
-        dataset = concatenate_datasets([a1, a2])
+        dataset = datasets.Dataset.from_list(list_of_data)
 
     dataset.save_to_disk("reading_comprehension_dataset")  # TODO: change name from
 
-    del dataset, a1, a2  # TODO: change name
+    # del dataset # TODO: change name
 
     train_generator(
```

Comment:
I guess we are using deep-speed during the prod training ?

Reply:
Oh sorry this is just removing some `del` statements and adding a comment.  

To have it automatically deleted, you could factor out a separate function:

```
def write_dataset(list_of_data, dataset_name):
  dataset = datasets.Dataset.from_list(list_of_data)
  dataset.save_to_disk(dataset_name)
```

and in the caller:

```
write_dataset(list_of_data, "reading_comprehension_dataset")
train_generator(
        model_name=model_name,
        dataset_name="reading_comprehension_dataset",
        .. etc
)
```

As soon as `write_dataset()` returns, the `dataset` local variable will go out of scope and be GC'd.

Reply:
Also I just noticed something suspect in this code block:

```
    if comprehension_type == SynthMode.BOTH:
        dataset = datasets.Dataset.from_list(list_of_data)

    dataset.save_to_disk("reading_comprehension_dataset")  # TODO: change name from
```

I don't see `dataset` ever being created if `if comprehension_type != SynthMode.BOTH`, and it looks like that code will throw an exception unless it is set to `SynthMode.BOTH`